### PR TITLE
Enable saving run stats from low-level API

### PR DESCRIPTION
**Pull request by alonre24**
- Refactor `RAI_RunStats` object and API
- Save stats entry in model/script struct
- Move stats update to background worker (at DAG post-process) from main thread (in reply callback)
- Add tests for stats from LLAPI. gears LLAPI, comprehensive synchronisation tests.

This also include some refactor of the `run_test_multiproc` util, that now allows every sub-process that represent a client to have a unique id (to enable some communications between the parent and children)

### CODE REVIEW

File: src/redis_ai_objects/stats.h [hunk 1]
```diff
@@ -34,66 +32,69 @@ mstime_t mstime(void);
...
+ * @param run_stats_entry RunStats entry pointer to store.
+ */
+void RAI_StatsStore(RedisModuleString *key, RAI_RunStats *run_stats_entry);
+
+/**
+ * @brief: Removes the statistical entry with the provided unique stats identifier
+ * @param info_key
  */
-void RAI_RemoveStatsEntry(void *infokey);
+void RAI_StatsRemoveEntry(void *info_key);
```
Comment (alonre24) [inline src/redis_ai_objects/stats.h:59]: Change void* to RMString

File: tests/flow/tests_commands.py [hunk 2]
```diff
@@ -358,3 +361,100 @@ def test_pytorch_scriptexecute_variadic_errors(env):
 
     check_error(env, con, 'AI.SCRIPTEXECUTE', 'ket{$}', 'bar_variadic', 'KEYS', 1 , '{$}', 'INPUTS', 'OUTPUTS')
 
+
+@with_test_module
+def test_ai_info_multiproc(env):
```
Comment (DvirDukhan) [inline tests/flow/tests_commands.py:366]: please re arrange this function to be more readable

File: src/redis_ai_objects/script.c [hunk 3]
```diff
@@ -56,9 +56,15 @@ void RAI_ScriptFree(RAI_Script *script, RAI_Error *err) {
...
     RedisModule_FreeString(NULL, script->tag);
-
-    RAI_RemoveStatsEntry(script->infokey);
-
+    // If the run stats which is stored under this key is the same one that the model holds a
+    // reference to, remove the entry from the global statistics dictionary as well. Otherwise,
+    // this key has been overwritten - just release the old run stats.
+    RAI_RunStats *stats = RAI_StatsGetEntry(script->info->key);
+    if (stats == script->info) {
+        RAI_StatsRemoveEntry(stats->key);
```
Comment (DvirDukhan) [inline src/redis_ai_objects/script.c:64]: who is the owner of the stats struct?

File: tests/flow/tests_gears_llapi.py [hunk 4]
```diff
@@ -190,27 +207,44 @@ async def ScriptRun_AsyncRunError(record):
         self.env.assertEqual(ret, b'OK')
         ret = con.execute_command('AI.TENSORSET', 'b{1}', 'FLOAT', 2, 2, 'VALUES', 2, 3, 2, 3)
         self.env.assertEqual(ret, b'OK')
+        self.info = info_to_dict(con.execute_command('AI.INFO', 'myscript{1}'))   # Runtime stats reported by AI.INFO.
```
[the remark this answers is not in the document]
Comment (alonre24) [reply tests/flow/tests_gears_llapi.py:210]: added

File: tests/flow/tests_gears_llapi.py [hunk 5]
```diff
@@ -99,34 +99,51 @@ async def ModelRun_AsyncRunError(record):
 
         con.execute_command('AI.TENSORSET', 'a{1}', 'FLOAT', 2, 2, 'VALUES', 2, 3, 2, 3)
         con.execute_command('AI.TENSORSET', 'b{1}', 'FLOAT', 2, 2, 'VALUES', 2, 3, 2, 3)
+        self.info = info_to_dict(con.execute_command('AI.INFO', 'm{1}'))   # Runtime stats reported by AI.INFO.
```
[the remark this answers is not in the document]
Comment (alonre24) [reply tests/flow/tests_gears_llapi.py:102]: added

File: tests/flow/tests_commands.py [hunk 6]
```diff
@@ -358,3 +361,179 @@ def test_pytorch_scriptexecute_variadic_errors(env):
...
+    if not TEST_TF:
+        env.debugPrint("Skipping test since TF is not available", force=True)
+        return
+    con = get_connection(env, '{1}')
+
+    # Load model protobuf and store its input tensors.
+    test_data_path = os.path.join(os.path.dirname(__file__), 'test_data')
+    model_filename = os.path.join(test_data_path, 'graph.pb')
+    with open(model_filename, 'rb') as f:
+        model_pb = f.read()
```
[the remark this answers is not in the document]
Comment (alonre24) [reply tests/flow/tests_commands.py:429]: right!

**Pull request merged**